keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/ProposalApplicationStrategy.java

```diff
@@ -0,0 +1,21 @@
+package io.improbable.keanu.algorithms.mcmc;
+
+import io.improbable.keanu.algorithms.mcmc.proposal.Proposal;
+import io.improbable.keanu.algorithms.variational.optimizer.Variable;
+
+import java.util.Set;
+
+public interface ProposalApplicationStrategy {
```

Comment:
I wonder if we should rename this? We are using the word `apply` for both a proposal and sampling algorithms.

Reply:
I don't think there's anything wrong with the interface name `ProposalApplicationStrategy`, i.e. it is the strategy for what to do when a proposal is applied.

We could rename the method `ProposalApplicationStrategy#apply`.

Unless you mean something else?

Reply:
Yeah, the method name change could also do